src/core/public/http/loading_count_service.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { BehaviorSubject, Observable, Subject } from 'rxjs';
+import {
+  distinctUntilChanged,
+  endWith,
+  map,
+  pairwise,
+  startWith,
+  takeUntil,
+  tap,
+} from 'rxjs/operators';
+import { FatalErrorsSetup } from '../fatal_errors';
+import { CoreService } from '../../types';
+
+/** @public */
+export interface LoadingCountSetup {
+  /**
+   * Adds a new source of loading counts. Used to show the global loading indicator when sum of all observed counts are
+   * more than 0.
+   * @param countSource$ an Observable to subscribe to for loading count updates.
+   */
+  addLoadingCount(countSource$: Observable<number>): void;
```

Comment:
ultraNIT: I think `addLoadingCounter` might be a little more explicit maybe? Or `addLoadingCountSource`, which matches the jsdoc?